cartridges/int_algolia/cartridge/scripts/algolia/customization/productModelCustomizer.js

```diff
@@ -54,7 +64,39 @@ function customizeProductModel(productModel) {
 }
 
 /**
- * Customize Localized Algolia Product.
+ * Create a hierarchical category Algolia object from a categories flat tree, for a localized model.
+ * For example, the following categories flat tree:
+ *   [
+ *     { id: 'newarrivals-electronics', name: 'Electronics' },
+ *     { id: 'newarrivals', name: 'New Arrivals' }
+ *   ]
+ * Will generate:
+ *   {
+ *      level_0: 'New Arrivals',
+ *      level_1: 'New Arrivals > Electronics'
+ *   }
+ * @param {Array} category - array of categories tree
+ * @returns {Object} - secondary category Algolia object
+ */
+function createAlgoliaLocalizedCategoryObject(category) {
+    var CATEGORIES_LEVEL_PREFIX = 'level_';
+    var result = {};
+
+    category
+        .concat()
+        .reverse()
+        .reduce(function (parentCategory, childCategory, index) {
```

Comment:
Some of these array methods [might not work with older compatibility modes](https://developer.salesforce.com/docs/commerce/b2c-commerce/guide/b2c-compatibility-mode-considerations.html#212). SFCC started supporting some ES6 elements only as late as 21.2

Reply:
Indeed, then I guess not much people use them because it has been implemented like that since the beginning.
This field is not documented actually, we should check it even worth keeping it. So I'll merge as-is to just fix the main bug for now, thanks for flagging.